packages/babel-core/src/transformation/normalize-opts.js

```diff
@@ -14,7 +14,7 @@ export default function normalizeOptions(config: ResolvedConfig): {} {
     moduleRoot,
     sourceRoot = moduleRoot,
 
-    sourceFileName = filenameRelative,
+    sourceFileName = path.basename(filenameRelative),
```

Comment:
Taking the basename here is not right. The sourceFileName _should_ include at least part of the file path.

Perhaps what we should do is change

    filenameRelative = filename || "unknown",

to

    filenameRelative = typeof filename === "string" ? path.relative(config.options.cwd, filename) || "unknown",

I'd also probably be in favor of removing `unknown` in favor of `null` but that could probably wait.

Reply:
Ok that sounds right, taking the actual relative path?

Reply:
Yeah I think that'd be ideal. We didn't have `cwd` in 6.x so it wasn't as nice of an option. Once we land the config PR it might even make sense to use the `root` value, but cwd works for now.

Reply:
Gotcha, that makes sense. will check if browserify+babelify deals well with that too and update tonight.